src/default.rs

```diff
@@ -17,6 +17,16 @@ thread_local! {
     static HANDLE: Handle = COLLECTOR.handle();
 }
 
+/// Returns the default handle.
+///
+/// # Safety
+///
+/// It should not be called in another TLS storage's `drop()`, because `HANDLE` may already be
+/// droped.
+pub unsafe fn default_handle() -> &'static Handle {
+    &*HANDLE.with(|handle| handle as *const _)
```

Comment:
The documentation for `with` is a bit ambiguous, but looking at [the source](https://doc.rust-lang.org/src/std/thread/local.rs.html#277-281) it should always panic if the key has been destroyed. Since we return a reference, it can't escape its scope and so `HANDLE` won't be destroyed while it exists. Hence, this _seems_ safe (but I could be wrong of course).

Reply:
It's unsafe because this function allows you to obtain a `&'static Handle` reference and then use it even after `HANDLE` gets destroyed, without any additional checks.

Reply:
@Vtec234 Destruction in the context of TLS is somewhat different than destruction in the rest of Rust unfortunately. When a thread exits, all of the registered TLS variables get destroyed - it doesn't have to do with scope. The problem arises when the `Drop` implementation for a TLS variable accesses another TLS variable which has already been dropped. @ezrosent and I have run into this issue, for example, when our own allocator caches (stored in TLS) were dropped after Crossbeam's TLS was dropped, but we relied on Crossbeam in order to implement `Drop`.

Reply:
In that case another option would be to forward `with` and expose something like `with_default_handle<F: FnOnce(&Handle) -> R>(f: F)`.

Reply:
That's true, but that might lead to callback spaghetti. It's just as easy to do `default_handle() -> Option<Handle>`, and somewhat more ergonomic.

Reply:
@joshlf But how would that work?

Handle is defined like this:

```rust
pub struct Handle {
    global: Arc<Global>,
    local: Local,
}
```

How would you return an `Option<Handle>`? You can't really clone it.

Reply:
Either `Option<Handle>` or `Handle` (since `with` panics on destroyed TLS, `Option` would probably always be `Some`) can be returned by cloning (#26) the default handle, but that of course has an associated overhead.

Reply:
@Vtec234 I believe the intention is to return the current thread's handle, not a new one. Cloning registers a brand new handle in the collector.

Reply:
Sorry, I meant to write `Option<&Handle>` instead of `Option<Handle>`. I've updated my original comment.

Reply:
And what's the issue with lifetimes when returning `Option<&Handle>`? Or is the lifetime problem only an issue when returning `Option<Handle>` (ie, not a reference)?

Reply:
@joshlf: Without a lifetime-binding argument (like in `with`) the lifetime of `&Handle` will have to be `'static`, and that can in turn be leaked until after the TLS drops since Rust allows returning `'static` references into scopes encompassing the function call.

Reply:
Ah I see. Fair enough.

Reply:
A problem with `fn default_handle() -> Option<&Handle>` is we cannot implement it in Rust stable.  I think we need to use [`HANDLE.try_with()`](https://doc.rust-lang.org/std/thread/struct.LocalKey.html#method.try_with), which is provided only in Rust nightly.

Reply:
Ah good point. Would you guys be cool with just having it be feature-gated on nightly?